packages/office-addin-usage-data/test/test.ts

```diff
@@ -162,23 +161,29 @@ describe("Test office-addin-usage data-package", function() {
     it("should parse error file paths with slashs", () => {
       addInUsageData.setUsageDataOff();
       const compareError = new Error();
-      compareError.name = "TestData-test";
-      compareError.message = "this error contains a file path:C:\\index.js";
+      compareError.name = "Error";
+      compareError.message = "this error contains a file path:index.js";
       // may throw error if change any part of the top of the test file
-      compareError.stack = "this error contains a file path:C:\\.js";
-      addInUsageData.maskFilePaths(err);
-      assert.equal(compareError.name, err.name);
-      assert.equal(compareError.message, err.message);
-      assert.equal(err.stack.includes(compareError.stack), true);
+      compareError.stack = "this error contains a file path:index.js";
+      const error = new Error(`this error contains a file path:C:/Users/admin/AppData/Roaming/npm/node_modules/alanced-match/index.js`);
+
+      addInUsageData.maskFilePaths(error);
+
+      assert.strictEqual(compareError.name, error.name);
+      assert.strictEqual(compareError.message, error.message);
+      assert.strictEqual(error.stack.includes(compareError.stack), true);
     });
     it("should parse error file paths with backslashs", () => {
       addInUsageData.setUsageDataOff();
       const compareErrorWithBackslash = new Error();
-      compareErrorWithBackslash.message = "this error contains a file path:C:\\excel file .xlsx";
-      compareErrorWithBackslash.stack = "this error contains a file path:C:\\.xlsx";;
+      compareErrorWithBackslash.message = "this error contains a file path:excel file .xlsx";
+      compareErrorWithBackslash.stack = "this error contains a file path:excel file .xlsx";;
+      const errWithBackslash = new Error(`this error contains a file path:C:\\Users\\admin\\AppData\\Local\Temp\\excel file .xlsx`);
+      
       addInUsageData.maskFilePaths(errWithBackslash);
-      assert.equal(compareErrorWithBackslash.message, errWithBackslash.message);
-      assert.equal(errWithBackslash.stack.includes(compareErrorWithBackslash.stack), true);
+
+      assert.strictEqual(compareErrorWithBackslash.message, errWithBackslash.message);
+      assert.strictEqual(errWithBackslash.stack.includes(compareErrorWithBackslash.stack), true);
```

Comment:
Why all these test changes?  The clone change shouldn't have required test changes.

Reply:
Yeah, it shouldn't, but the `mask` function was bugged and wasn't actually masking the files consistently.
The syntax of having:
`regexRemoveAbsoluteUserFilePathsFromStack` and  `regexRemoveFirstFilePathFromStack` were probably a way to make the tests pass but were not really correct.

Reply:
Can you explain why you think it was made that way to make the tests pass and why it wasn't correct and why this new way is correct?

Reply:
I didn't dive deep to understand why it wasn't correct, it simply didn't get the regex and parse user files. 

The process I made to figure out that there was a problem with the function: I removed side effects on the functions that call it and on the tests -> this made the tests start to fail. So my guess is that the function was only working when the tests suites were the way they were. 

Also, I think the way the code looks now is more "right". We want to find user paths on the code and remove them. Why can't we just have one regex for that? My guess is that the code before was overly complicated to account for the side effects in the tests.